### PR TITLE
SNMP base data

### PR DESCRIPTION
**Pull request by martinlindhe**
As a frontend developer, I want to create a user-friendly UI to enable customers to monitor devices via SNMP without having any knowledge about SNMP. The user does not need to know anything about OIDs. By selecting a preset for the most common use-cases the right OIDs are queried or discovered.
In a first version (POC) we prove the complete chain.

    requesting a check
    connect to the device
    query data
    send back results

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -11,21 +16,17 @@
   version = "2.5.1"
 
 [[constraint]]
+  branch = "master"
   name = "github.com/kardianos/service"
-  revision = "4c239ee84e7bb93441b1b6a3f2db62d40e0e6cbd"
 
 [[constraint]]
+  branch = "master"
```
[the remark this answers is not in the document]
Comment (martinlindhe) [reply Gopkg.toml]: Sure, but what would it accomplish?

We carry vendor folder and Gopkg.lock to lock on specific revisions already

File: Gopkg.toml [hunk 2]
```diff
@@ -11,21 +16,17 @@
   version = "2.5.1"
 
 [[constraint]]
+  branch = "master"
```
[the remark this answers is not in the document]
Comment (martinlindhe) [reply Gopkg.toml]: Fixed

File: Gopkg.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (martinlindhe) [reply Gopkg.toml]: Fixed

File: Gopkg.toml [hunk 3]
```diff
@@ -35,8 +36,8 @@
   version = "1.2.2"
 
 [[constraint]]
+  branch = "master"
```
[the remark this answers is not in the document]
Comment (martinlindhe) [reply Gopkg.toml]: Fixed

File: snmp.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+		return m, fmt.Errorf("connect err: %v", err)
+	}
+	defer params.Conn.Close()
+
+	oids, err := presetToOids(check.Preset)
+	if err != nil {
+		return m, err
+	}
+
+	result, err := params.GetBulk(oids, uint8(len(oids)), 255)
```
[the remark this answers is not in the document]
Comment (martinlindhe) [reply snmp.go]: Fixed

File: handler.go [hunk 5]
```diff
@@ -690,10 +689,50 @@ func (fm *Frontman) processInput(input *Input, resultsChan chan<- Result) {
 		}(check)
 	}
 
+	for _, check := range input.SNMPChecks {
+		wg.Add(1)
+		go func(check SNMPCheck) {
+			defer wg.Done()
+
+			if check.UUID == "" {
```
Comment (troian) [inline handler.go]: Is it worth to use some external package to wrap UUID? It could validate for example it's actually UUID not just a string

Comment (martinlindhe) [reply handler.go]: The property called "UUID" here is just a string, see "checkUUID" in https://github.com/cloudradar-monitoring/frontman/blob/master/example.json#L3

Comment (martinlindhe) [reply handler.go]: Also, this whole function was based on existing functions, see 

https://github.com/cloudradar-monitoring/frontman/blob/master/handler.go#L620

and

https://github.com/cloudradar-monitoring/frontman/blob/master/handler.go#L656

**Pull request merged**